### ISSUE TITLE
stdint.h included in SuperLU header files, but does not exist for Visual C++ 9.

### ISSUE THREAD

**Issue by matthew-brett** [scipy/scipy#7551]
I am trying to compile Scipy on Windows, using Visual C++ 9 (Python 2.7) (and mingwpy gfortran).

I get the following error:

```
...
creating build\temp.win32-2.7\scipy\sparse\linalg\dsolve\SuperLU\SRC
C:\Program Files (x86)\Common Files\Microsoft\Visual C++ for Python\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG /arch:SSE2 -DNO_TIMER=1 -DUSE_VENDOR_BLAS=1 -Iscipy\sparse\linalg\dsolve\SuperLU\SRC -Ic:\tmp\venv\lib\site-packages\numpy\core\include /Tcscipy\sparse\linalg\dsolve\SuperLU\SRC\ccolumn_bmod.c /Fobuild\temp.win32-2.7\scipy\sparse\linalg\dsolve\SuperLU\SRC\ccolumn_bmod.obj
ccolumn_bmod.c
c:\repos\scipy\scipy\sparse\linalg\dsolve\superlu\src\slu_cdefs.h(83) : fatal error C1083: Cannot open include file: 'stdint.h': No such file or directory
error: Command "C:\Program Files (x86)\Common Files\Microsoft\Visual C++ for Python\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG /arch:SSE2 -DNO_TIMER=1 -DUSE_VENDOR_BLAS=1 -Iscipy\sparse\linalg\dsolve\SuperLU\SRC -Ic:\tmp\venv\lib\site-packages\numpy\core\include /Tcscipy\sparse\linalg\dsolve\SuperLU\SRC\ccolumn_bmod.c /Fobuild\temp.win32-2.7\scipy\sparse\linalg\dsolve\SuperLU\SRC\ccolumn_bmod.obj" failed with exit status 2
```

I think these arrived with commit 1fabda5 which updated our copy of SuperLU.

I worked round it with:

```
curl -L https://raw.githubusercontent.com/mattn/gntp-send/master/include/msinttypes/stdint.h -o scipy/sparse/linalg/dsolve/SuperLU/SRC/stdint.h
```

https://stackoverflow.com/questions/126279/c99-stdint-h-header-and-ms-visual-studio#126285

Maybe there should be some conditional guard / fallback in those def files for the possibility of the missing file.

**Comment by rgommers** [comment]
> using Visual C++ 9 (Python 2.7) (and mingwpy gfortran).

Has that ever worked for you??

**Comment by matthew-brett** [comment]
Actually, I have not tried that combination before.  But - it might just work.  We were speculating that, as for OpenBLAS, there may be no Fortran code that uses objects that can't be passed across VC runtimes - such as file objects.   Or, it might be possible to move any such code into C or Python.  In which case we would have solved the problem of compiling Scipy for Windows.

**Comment by pv** [comment]
stdint.h is in c99 standard, which microsoft is dragging its feet in implementing.
To the second question: I don't think there is code in scipy that passes CRT resources between fortran and C (mainly because these objects don't really exist in F77).

**Comment by pv** [comment]
fix in gh-7554

**Comment by pv** [comment]
Did the MSVC+gfortran combination end up working?

**Comment by matthew-brett** [comment]
@carlk and I are currently iterating over it.  The complication is trying to get the gfortran outputs linked with the VC++ outputs - including a mix of VC and mingw libraries.  I'll try and put up an Appveyor build with progress so far.

**Comment by matthew-brett** [comment]
Here's the appveyor build, with the current error:

https://ci.appveyor.com/project/matthew-brett/build-scipy/build/1.0.5#L4401

Any suggestions gratefully received.

**Comment by matthew-brett** [comment]
Patching out the mingwex.lib and mingw32.lib from the library flags gets me a little further:

https://ci.appveyor.com/project/matthew-brett/build-scipy/build/1.0.9#L4430

```
fftpack.lib(rfftb1.o) : error LNK2001: unresolved external symbol _sinf
```

@carlkl - any thoughts?

**Comment by pv** [comment]
_sinf is c99 symbol, so MSVC maybe doesn't have it. I suppose these 
symbols are generated by the gfortran compiler, which assumes they are 
present. Maybe it's possible to add C file with stubs along the lines 
'float sinf(float x) { return sin(x); }', and link it in only when using 
MSVC.

**Comment by carlkl** [comment]
Maybe using https://github.com/pydata/numexpr/blob/master/numexpr/win32/stdint.h could help there.

I presume for this case a static library `mingw.lib` including all missing symbols as well as all gcc dependencies has to be created first of all.

**Comment by matthew-brett** [comment]
Sorry to keep appending to this issue (stdint.h) which Pauli solved with #7554 .  Closing this one in favor of a new MSVC / gfortran build issue at #7597.
